File: src/ortools/OrTools.Optimizer/Models/Tasks/RepeatingSchedule.cs
namespace OrTools.Optimizer.Models.Tasks;

public record RepeatingSchedule
{
    public int? MinDayCount { get; init; }
    public int OptDayCount { get; init; } = 1;
    public int? MinWeekCount { get; init; }
    public int OptWeekCount { get; init; } = 1;
}

File: src/timefold/src/main/java/com/pso/timefold/domain/RepeatingSchedule.java
package com.pso.timefold.domain;

import com.fasterxml.jackson.annotation.JsonSetter;
import com.fasterxml.jackson.annotation.Nulls;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor
public class RepeatingSchedule {

    private int minDayCount;

    @JsonSetter(nulls = Nulls.SKIP)
    private int optDayCount = 1;

    private int minWeekCount;

    @JsonSetter(nulls = Nulls.SKIP)
    private int optWeekCount = 1;
}
